backend/compact-connect/docs/README.md

```diff
@@ -35,12 +35,12 @@ For your convenience, use of this feature is included in the [Postman Collection
 
```

Comment:
# General Comments & Questions

* The frontend does not display an error when the user is unauthorized to upload to a particular jurisdiction. @jsandoval81 
* The frontend does not restrict the bulk upload jurisdiction list to those jurisdictions to which I have write permissions. @jsandoval81 
* How do I set up authentication / Cognito with a sandbox deployment? I'm getting errors when I use the test app callback URL. @jusdino 
* How would you recommend running pa11y or ZAP automated testing now that the main pages are authenticated? @jusdino

Reply:
> How do I set up authentication / Cognito with a sandbox deployment? I'm getting errors when I use the test app callback URL. @jusdino

There is a script under `backend/compact-connect/bin` to create yourself a user. After you've run that, you should be able to log in. The next thing would be making sure that the Cognito app client is configured to allow the callback to your UI. That part depends on where you're running the UI. If you're running the UI from CloudFront with a domain name, it should work. If you're running the UI locally, you'll need to make sure that you have the context variable `"allow_local_ui": true` set for your sandbox environment (included in `cdk.context.sandbox-example.json`. That flag will allow authorization callbacks to `"http://localhost:3018/auth/callback"`.

> How would you recommend running pa11y or ZAP automated testing now that the main pages are authenticated? @jusdino

Each of those tools supports authentication in their own way - here are a couple of references that offer a hint:
- https://www.zaproxy.org/docs/desktop/start/features/authentication/
- https://github.com/pa11y/pa11y/issues/242

If you think it's time for us to automate those scans, we should probably talk to @isabeleliassen about getting some tickets in the backlog. We can also talk about disabling MFA requirements in `test` (or your sandbox), if MFA is getting in the way of efficient testing for you.

> When I test on additional browsers, Cognito gives me an authentication error. @jusdino

It looks like you're running afoul of Cognito's intrusion detection / advanced security monitoring. If Cognito keeps shutting you out, like MFA, we can talk about loosening up Cognito's security configuration for a pre-production environment.

Reply:
> > How do I set up authentication / Cognito with a sandbox deployment? I'm getting errors when I use the test app callback URL. @jusdino
> 
> There is a script under `backend/compact-connect/bin` to create yourself a user. After you've run that, you should be able to log in. The next thing would be making sure that the Cognito app client is configured to allow the callback to your UI. That part depends on where you're running the UI. If you're running the UI from CloudFront with a domain name, it should work. If you're running the UI locally, you'll need to make sure that you have the context variable `"allow_local_ui": true` set for your sandbox environment (included in `cdk.context.sandbox-example.json`). That flag will allow authorization callbacks to `"http://localhost:3018/auth/callback"`.
> 
Helpful, thank you! I'll try this and return with any follow-ups.

> > How would you recommend running pa11y or ZAP automated testing now that the main pages are authenticated? @jusdino
> 
> Each of those tools supports authentication in their own way - here are a couple of references that offer a hint:
> 
> * https://www.zaproxy.org/docs/desktop/start/features/authentication/
> * [pa11y authentication and login pa11y/pa11y#242](https://github.com/pa11y/pa11y/issues/242)
> 
> If you think it's time for us to automate those scans, we should probably talk to @isabeleliassen about getting some tickets in the backlog. We can also talk about disabling MFA requirements in `test` (or your sandbox), if MFA is getting in the way of efficient testing for you.
> 

Right, I read through a lot of this but was having trouble setting up and didn't want to spend too much time on it. I think partly it can't easily deal with MFA, so perhaps disabling MFA in `test` or my sandbox is a good first step. North star is still to automate these scans, of course, but if we can't get them running manually, I doubt we'll be able to get them running automatically anyways.

> > When I test on additional browsers, Cognito gives me an authentication error. @jusdino
> 
> It looks like you're running afoul of Cognito's intrusion detection / advanced security monitoring. If Cognito keeps shutting you out, like MFA, we can talk about loosening up Cognito's security configuration for a pre-production environment.

Got it. Definitely a blocker as I'm unable to test on various browsers. I'm inclined to prefer identical setups on prod and pre-prod, but if this is the only reasonable option, then so be it. FWIW, I didn't even get to the MFA screen when I received these errors, so it feels like something that's happening _before_ MFA.